activerecord/lib/active_record/connection_adapters/mysql2/database_statements.rb

```diff
@@ -98,12 +98,13 @@ def with_multi_statements
           end
 
           def raw_execute(sql, name, async: false, allow_retry: false, materialize_transactions: true)
-            log(sql, name, async: async) do
+            log(sql, name, async: async) do |notification_payload|
               with_raw_connection(allow_retry: allow_retry, materialize_transactions: materialize_transactions) do |conn|
                 sync_timezone_changes(conn)
                 result = conn.query(sql)
                 verified!
                 handle_warnings(sql)
+                notification_payload[:row_count] = result&.size || 0
```

Comment:
Can `result` actually be nil here (or in any of the other cases)?

Reply:
When running the tests I had many cases of result being nil. I did not verify this for every location individually, but got enough cases of it happening that my preference was to be defensive here. 

```
/workspaces/rails/activerecord/lib/active_record/connection_adapters/mysql2/database_statements.rb:107:in `block (2 levels) in raw_execute': undefined method `size' for nil:NilClass (NoMethodError)

                notification_payload[:row_count] = result.size || 0
                                                         ^^^^^
```

I just tested it for this specific statement and got a failure immediately

Reply:
I ran the test suite for all adapters and mysql2 seems to be the only one where result can be `nil`. I'd prefer not to add safe navigation unless there's evidence that it's needed.